dwave/optimization/symbols.pyx

```diff
@@ -794,6 +802,54 @@ cdef class Constant(ArraySymbol):
         buffer.strides = <Py_ssize_t*>(self.ptr.strides().data())
         buffer.suboffsets = NULL
 
+    def __index__(self):
+        if not self._is_integer():
+            # Follow NumPy's error message
+            # https://github.com/numpy/numpy/blob/66e1e3/numpy/_core/src/multiarray/number.c#L833
+            raise TypeError("only integer scalar constants can be converted to a scalar index")
+
+        return <Py_ssize_t>deref(self.ptr.buff())
+
+    def __richcmp__(self, rhs, int op):
+        # __richcmp__ is a special Cython method
+
+        # If rhs is another Symbol, defer to ArraySymbol to handle the
+        # operation. Which may or may not actually be implemented.
+        # Otherwise, defer to NumPy.
+        # We could also check if rhs is another Constant and handle that differently,
+        # but that might lead to confusing behavior so we treat other Constants the
+        # same as any other symbol.
+        lhs = super() if isinstance(rhs, ArraySymbol) else np.asarray(self)
+
+        if op == cpython.object.Py_EQ:
+            return lhs.__eq__(rhs)
+        elif op == cpython.object.Py_GE:
+            return lhs.__ge__(rhs)
+        elif op == cpython.object.Py_GT:
+            return lhs.__gt__(rhs)
+        elif op == cpython.object.Py_LE:
+            return lhs.__le__(rhs)
+        elif op == cpython.object.Py_LT:
+            return lhs.__lt__(rhs)
+        elif op == cpython.object.Py_NE:
+            return lhs.__ne__(rhs)
+        else:
+            return NotImplemented  # this should never happen, but just in case
```

Comment:
```suggestion
        return NotImplemented  # this should never happen, but just in case
```

Reply:
I think the nesting makes it clearer that it's because those are the only options.